src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                {
+                    // Executes syntax building methods from bottom to the top of the tree.
+                    statement = processingState.ExecuteMethods().WithAdditionalAnnotations(Simplifier.Annotation);
+                    return true;
+                }
+
+                statement = default;
+                return false;
+            }
+
+            private bool TryProcessQueryBody(
+                QueryBodySyntax queryBody,
+                ProcessingState processingState)
+            {
+                foreach (var queryClause in queryBody.Clauses)
+                {
+                    if (!TryProcessQueryClause(queryClause, processingState))
+                    {
+                        return false;
+                    }
+                }
+
+                // GroupClause is not supported by the conversion
+                return queryBody.SelectOrGroup is SelectClauseSyntax selectClause &&
+                    TryProcessSelectClause(selectClause, queryBody.Continuation, processingState);
+            }
+
+            private bool TryProcessQueryClause(
+                QueryClauseSyntax queryClause,
+                ProcessingState processingState)
+            {
+                switch (queryClause.Kind())
+                {
+                    case SyntaxKind.WhereClause:
+                        return TryProcessWhereClause((WhereClauseSyntax)queryClause, processingState);
+                    case SyntaxKind.LetClause:
+                        return TryProcessLetClause((LetClauseSyntax)queryClause, processingState);
+                    case SyntaxKind.FromClause:
+                        return TryProcessFromClause((FromClauseSyntax)queryClause, processingState);
+                    case SyntaxKind.JoinClause:
+                        return TryProcessJoinClause((JoinClauseSyntax)queryClause, processingState);
+                }
+
+                // OrderBy is not supported by foreach.
+                return false;
+            }
+
+            private bool TryProcessSelectClause(
+                SelectClauseSyntax selectClause,
+                QueryContinuationSyntax queryContinuation,
+                ProcessingState processingState)
+            {
+                // TODO semantic check for selectClause.Expression
+                // End of the queue
+                if (queryContinuation == null)
+                {
+                    processingState.AddStatementProcessingMethod(
+                        s => WrapWithBlock(processingState.LeafExpressionCreationMethod(selectClause.Expression)));
+                    return true;
+                }
+                else
+                {
+                    SymbolInfo symbolInfo = _semanticModel.GetSymbolInfo(queryContinuation, _cancellationToken);
+                    if (symbolInfo.Symbol == null ||
+                        symbolInfo.Symbol is IMethodSymbol methodSymbol &&
```

Comment:
>symbolInfo.Symbol is IMethodSymbol methodSymbol [](start = 24, length = 47)

Could you please clarify what method are we checking for here?

Reply:
Also, why are we not checking its name?

---
In reply to: [177187123](https://github.com/dotnet/roslyn/pull/25362#discussion_r177187123) [](ancestors = 177187123)

Reply:
agreed to skip verification of query clauses for now

---
In reply to: [177188337](https://github.com/dotnet/roslyn/pull/25362#discussion_r177188337) [](ancestors = 177188337,177187123)